app/models/registration.rb

```diff
@@ -67,21 +67,26 @@ def user_in_course
   # my-start >= my-exam-start
   # my-end <= my-exam-end
   # my-end <= my-start + my-duration
+  #
+  # NOTE: If the exam version itself defines a start time/end time/duration,
+  # those override the exam-specified times.
 
   def accommodated_start_time
-    accommodation&.new_start_time || exam.start_time
+    accommodation&.new_start_time || exam_version.effective_start_time
   end
 
   def accommodated_duration
-    exam.duration * (accommodation&.factor || 1)
+    exam_version.effective_duration * (accommodation&.factor || 1)
   end
 
   def accommodated_extra_duration
-    accommodated_duration - exam.duration
+    accommodated_duration - exam_version.effective_duration
   end
 
   def accommodated_end_time
-    accommodated_start_time + (exam.end_time - exam.start_time) + accommodated_extra_duration
+    accommodated_start_time +
+      (exam_version.effective_end_time - exam_version.effective_start_time) +
```

Comment:
Inside the parens is `effective_time_window`?

Reply:
Yes, though that helper isn't defined anywhere. Iirc, it's fiddly to define that when dealing with a new_record, because it doesn't pull from the source fields correctly or something. I don't remember exactly.

Reply:
You can't use the one on `exam_version` that you added? https://github.com/CodeGrade/hourglass/pull/418/files#diff-3c8bded9ae6017e1076c4f90e2f5cbbd62fb7576deb0587e8f3c77969286bc8aR125-R127

Reply:
/facepalm